### PR TITLE
refactor(trie): move hashed post state database transaction related functions to `reth-trie-db` crate

### PR DESCRIPTION
**Pull request by RomanHodulak**
## About
A different variation of #9970 based on https://github.com/paradigmxyz/reth/pull/9970#issuecomment-2263020181

### CODE REVIEW

File: crates/trie/db/src/state.rs [hunk 1]
```diff
@@ -111,6 +120,13 @@ pub trait DatabaseStateRoot<'a, TX>: Sized {
     ) -> Result<(B256, TrieUpdates), StateRootError>;
 }
 
+/// Extends [`HashedPostState`] with operations specific for working with a database transaction.
+pub trait DatabaseHashedPostState<'a, TX>: Sized {
```
Comment (mattsse) [inline crates/trie/db/src/state.rs:124]: do we need this lifetime?

**Review by mattsse — COMMENTED** [review]
pending @rkrasiuk

**Review by rkrasiuk — APPROVED** [review]
looks good

File: crates/trie/db/src/state.rs (hunk 1 above)
Comment (RomanHodulak) [reply crates/trie/db/src/state.rs]: nah

**Pull request merged**